### DISCUSSION TITLE
DevExpress Blazor Components in Oqtane Module

### DISCUSSION THREAD

**Discussion by zoltanhalasz** [oqtane/oqtane.framework#1051] [Q&A]
Hello,

does anybody have a method to include components from DevExpress into Oqtane?
I found great instructions for Radzen and Syncfusion:
https://blazorhelpwebsite.com/ViewBlogPost/47
and
https://blazorhelpwebsite.com/ViewBlogPost/46
but DevExpress seems to be different for me.

**Comment by hishamco** [comment]
Why it's different?

**Comment by sbwalker** [comment]
@zoltanhalasz In general, component libraries will have assemblies and static resources - both of which will need to be deployed the appropriate location in Oqtane so that they are accessible. In addition, there will likely be some services which need to be registered in IServerStartup and links added for the static resources using IHostResources. @ADefWebserver's tutorials for Radzen and Syncfusion provide very detailed instructions. How is DevExpress different from the other component libraries?

**Reply by ADefWebserver** [reply]
> @ADefWebserver's tutorials for Radzen and Syncfusion provide very detailed instructions. How is DevExpress different from the other component libraries?

The pattern used in the **Syncfusion** tutorial should work for *all* component libraries. That tutorial is here: https://blazorhelpwebsite.com/ViewBlogPost/47

**Reply by zoltanhalasz** [reply]
I could say that for Syncfusion there are a small number of dll and javascript files to include in Oqtane. With DevExpress there are tens  of dlls and javascript files in the package, and I just couldn't figure out which one to include. But, no worries, I will figure it out as the project that I will work on, has this solved already.

**Reply by ADefWebserver** [reply]
@zoltanhalasz While the **Syncfusion** only had 4 .dlls it did have **137** JavaScript files! :)

**Reply by zoltanhalasz** [reply]
It's possible, but Devexpress has tens of dlls (which one to select?) and tens of Js files (with no specific naming to know which to select). I will come back with the specifics in the following days, since its night in Eastern Europe now :)
But don't worry, I will even have the answer in couple of days. Thanks really for the great material on your site.

**Comment by zoltanhalasz** [comment]
I studied the mentioned tutorials in detail and they work for me. In my company, we'll work for a client in Oqtane, who already implemented DevExpress so I will get the specific answer soon, if there is any difference. I appreciate  the answer from both of you.

**Reply by rchibnik** [reply]
I am trying to get this to work as well; I can't seem to find the nupkg - Devexpress only lets you install it, not download the nupkg so i can't extract any .js or css to load in HostResources.

Anyone have a link to the nupkg?  I can get a grid to display, partially styled using boostrap.css as dx-blazor.css but I get an Interop Exception: Could not find 'DxBlazor.Grid.init' ('DxBlazor' was undefined).

**Reply by rchibnik** [reply]
Found the nupkg: selecting the package in vs19 nuget package manager, upper right hand corner is a link to the nuget server and clicking it downloads. 

Getting syntax errors around import/export in the DevExpress.js;

**Reply by rchibnik** [reply]
Got this to work. Following the syncfusion tutorial worked.
https://blazorhelpwebsite.com/ViewBlogPost/47
Copied all the staticresources to _content;

Next step is to create a module template.

**Comment by abukres** [comment]
I spent a couple hours yesterday following @ADefWebserver Oqtane ebook. The last step I tried was replacing Syncfusion's Blazor grid with DevExpress' one and it displayed fine. I was using VS 2022. However the workflow wasn't fun. IIS Express always kept the Oqtane.Server.locked from being overwritten and everytime I made a module change, I had to unlock it which meant restarting the Oqtane app. It would be nice if the hot reload worked.

It was just a little attempt of proof of concept to see if DevExpress components would work in Oqtane.

**Reply by ADefWebserver** [reply]
If I start Oqtane with "Start without debugging", and only build the external module project, I can usually avoid that. However, not always.
